cmd/oras/internal/option/target.go

```diff
@@ -98,18 +98,18 @@ func (opts *Target) ApplyFlagsWithPrefix(fs *pflag.FlagSet, prefix, description
 func (opts *Target) Parse() error {
 	switch {
 	case opts.IsOCILayout:
+		var err error
 		opts.Type = TargetTypeOCILayout
 		if len(opts.headerFlags) != 0 {
 			return errors.New("custom header flags cannot be used on an OCI image layout target")
 		}
-		return nil
+		opts.Path, opts.Reference, err = opts.parseOCILayoutReference()
```

Comment:
Can we move the path and reference setup into the parse function?
```suggestion
		return opts.parseOCILayoutReference()
```